### PR TITLE
Fix for off by one error

### PR DESCRIPTION
**Pull request by princejwesley**
Fix for
~~1) `http://example.com/a/b/c/` -> `http://example.com/a/b/c`~~
2) `fn('http://example.com/a/b/cd', 24)` returns of length `25`

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ test(function (t) {
 	t.is(fn('http://sindresorhus.com/foo/bar/baz/faz', 30), 'http://sindresorhus.com/…/faz');
 	t.is(fn('http://example.com/a/cool/page/that-is-really-deeply/nested/', 40), 'http://example.com/…/nested/');
 	t.is(fn('http://example.com/a/b/c/d', 24).length, 24);
+	t.is(fn('http://example.com/a/b/c/d', 24).length, 24);
```
Comment (sindresorhus) [inline test.js]: This is identical to the above.

File: index.js [hunk 2]
```diff
@@ -14,6 +14,11 @@ module.exports = function (input, length) {
 		return input;
 	}
 
+	var strippedTrailingSlash = input.replace(/\/$/, '');
+	if (strippedTrailingSlash.length === length) {
+		return strippedTrailingSlash;
+	}
```
Comment (sindresorhus) [inline index.js]: Why should we do this? I don't get the point of this.

File: index.js [hunk 3]
```diff
@@ -23,8 +28,7 @@ module.exports = function (input, length) {
 
 	while (i--) {
 		var x = pathParts[i];
-
-		if (remainingLength < x.length) {
+		if (remainingLength < x.length + 1) {
```
Comment (sindresorhus) [inline index.js:26]: The existing behaviour is already tested: https://github.com/sindresorhus/truncate-url/pull/3/files#r39253808

File: index.js (hunk 2 above)
Comment (princejwesley) [inline index.js]: we can return complete url for this use case: `http://example.com/a/b/c/` -> `http://example.com/a/b/c`

File: index.js (hunk 3 above)
Comment (sindresorhus) [inline index.js:26]: Can you add a test the fails before this change and passes after?

Comment (princejwesley) [inline index.js:26]: `fn('http://example.com/a/b/cd', 24)` returns `http://example.com/…/b/cd` of length 25.
We need `x.length +1` because of the below code:

``` js
remainingLength -= x.length + 1;
```

Comment (princejwesley) [inline index.js:26]: [This PR](https://github.com/sindresorhus/truncate-url/pull/1/files#diff-168726dbe96b3ce427e7fedce31bb0bcL29) introduced this bug

Comment (princejwesley) [inline index.js:26]: Here is the [failed test](https://github.com/sindresorhus/truncate-url/pull/4)

File: index.js (hunk 2 above)
Comment (schnittstabil) [inline index.js]: I don't think so: `http://example.com/a/b/c/` and `http://example.com/a/b/c` are different Urls.

File: index.js (hunk 3 above)
Comment (schnittstabil) [inline index.js:26]: Nice catch, <s>would it be possible to add your failing test to this PR to document the changed behavior, instead of creating a new PR?</s>
I think @sindresorhus  means that

``` js
test('should pass corner case', function (t) {
    var result = fn('http://example.com/a/b/cd', 24);
    t.true(result.length <= 24, result + ' is > 24');
    t.end();
});
```

is already covered by:

``` js
t.is(fn('http://example.com/a/b/cd', 24), 'http://example.com/…/cd')`
```

(@sindresorhus: the latter currently fails)

File: index.js (hunk 2 above)
Comment (princejwesley) [inline index.js]: I'll remove it

**Comment by sindresorhus** [comment]
Thanks :)
